### PR TITLE
refactor(hostname): use clap instead of getopts for consistency 

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: src/uu/hostname/src/hostname.rs [hunk 1]
```diff
@@ -61,53 +66,58 @@ pub fn uumain(args: Vec<String>) -> i32 {
...
         }
-        _ => {
-            show_error!("{}", msg_wrong_number_of_arguments!(0, 1));
-            1
-        }
     }
 }
 
-fn display_hostname(matches: Matches) -> i32 {
+fn display_hostname(matches: ArgMatches) -> i32 {
```
Comment (rivy) [inline src/uu/hostname/src/hostname.rs:117]:
```suggestion
fn display_hostname(matches: &ArgMatches) -> i32 {
```

File: src/uu/hostname/src/hostname.rs [hunk 2]
```diff
@@ -61,53 +66,58 @@ pub fn uumain(args: Vec<String>) -> i32 {
...
-        0 => display_hostname(matches),
-        1 => {
-            if let Err(err) = xsethostname(matches.free.last().unwrap()) {
+                possible",
+        ))
+        .arg(Arg::with_name(OPT_HOST))
+        .get_matches_from(&args);
+
+    match matches.value_of(OPT_HOST) {
+        None => display_hostname(matches),
```
Comment (rivy) [inline src/uu/hostname/src/hostname.rs:105]:
```suggestion
        None => display_hostname(&matches),
```

Comment (rivy) [reply src/uu/hostname/src/hostname.rs:105]: Required for `v1.31.0`.

File: src/uu/hostname/src/hostname.rs (hunk 1 above)
Comment (rivy) [reply src/uu/hostname/src/hostname.rs:117]: ... for `v1.31.0`.

File: src/uu/hostname/src/hostname.rs (hunk 2 above)
Comment (sylvestre) [reply src/uu/hostname/src/hostname.rs:105]: merci !

**Comment by rivy** [comment]
I'm not sure that the we should generally change from `getopt` to `clap`.

I just looked at the code files generated on my Windows host machine. The simple, equivalent, change from `getopt` to `clap` more than doubled the individual utility size from `301k` to `651k`, without any functional change or reduction in technical debt. I don't think that's a good exchange.

If there is some needed functionality for a specific utility, I'm on-board... but I don't think the conversion should be done by reflex.

Your thoughts?

**Comment by sylvestre** [comment]
@rivy Excellent point. I didn't notice the increase of size.

However, the current argument management is a mess in coreutils. Each binary is doing it differently. 
Using a single way would be better for the future of the project. 
Now, clap seems to be the defacto standard in rust for argument management and can do (almost) everything. 
As you noticed, I focused first on program without any code coverage. I started the clap migration when I noticed how `date` was broken.

Also, there are some discussions upstream to decrease the size of the binary : https://github.com/clap-rs/clap/issues/1365 So, hopefully, it will be automagically fixed!

Last but not least, a rhetorical question, does it matter? :) I agree that it isn't clean but what is 350k?

**Pull request merged**